modules/http4s/test/src/smithy4s/http4s/RouterSpec.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.http4s
+
+import cats.effect.IO
+import cats.effect.syntax.resource._
+import smithy4s.example._
+import weaver._
+import org.http4s.client.Client
+import org.http4s.server.Router
+
+object RouterSpec extends SimpleIOSuite {
+
+  private val pizzaImpl: PizzaAdminService[IO] =
+    new PizzaAdminService.Default[IO](IO.stub) {
+      override def version(): IO[VersionOutput] =
+        IO.pure(VersionOutput("1.0.0"))
+    }
+
+  test(
+    "SimpleProtocolBuilder (client) fails when the protocol is not present"
+  ) {
+    val test = for {
+      routes <- SimpleRestJsonBuilder.routes(pizzaImpl).resource
+      routed = Router("/api" -> routes).orNotFound
+      client = Client.fromHttpApp(routed)
+      pizzaClient <- SimpleRestJsonBuilder(PizzaAdminService)
+        .client(client)
+        .resource
```

Comment:
AFAICS this can't really work because the server has that `/api` prefix, while the client doesn't use it. I'd try with `.client(client).uri(uri"/api")`.

Reply:
If you do that, then this change seems to fix it:

```diff
diff --git a/modules/http4s/src/smithy4s/http4s/SmithyHttp4sRouter.scala b/modules/http4s/src/smithy4s/http4s/SmithyHttp4sRouter.scala
index 47cc0ec9..e8cd6bcc 100644
--- a/modules/http4s/src/smithy4s/http4s/SmithyHttp4sRouter.scala
+++ b/modules/http4s/src/smithy4s/http4s/SmithyHttp4sRouter.scala
@@ -43,7 +43,7 @@ class SmithyHttp4sRouter[Alg[_[_, _, _, _, _]], Op[_, _, _, _, _], F[_]](
   val routes: HttpRoutes[F] = Kleisli { request =>
     for {
       endpoints <- perMethodEndpoint.get(request.method).toOptionT[F]
-      path = request.uri.path.segments.map(_.decoded()).toArray
+      path = request.pathInfo.segments.map(_.decoded()).toArray
       (endpoint, pathParams) <- endpoints.collectFirstSome(_.matchTap(path)).toOptionT[F]
       response <- OptionT.liftF(endpoint.httpApp(request.withAttribute(pathParamsKey, pathParams)))
     } yield response
```

but I'm not sure if there might be other implications.